network/codec/codes.go

```diff
@@ -147,81 +154,96 @@ func MessageCodeFromInterface(v interface{}) (uint8, string, error) {
 // of the message code represents.
 // Expected error returns during normal operations:
 //   - ErrUnknownMsgCode if message code does not match any of the configured message codes above.
-func InterfaceFromMessageCode(code uint8) (interface{}, string, error) {
+func InterfaceFromMessageCode(code MessageCode) (interface{}, string, error) {
 	switch code {
 	// consensus
 	case CodeBlockProposal:
-		return &messages.BlockProposal{}, "BlockProposal", nil
+		return &messages.BlockProposal{}, what(messages.BlockProposal{}), nil
```

Comment:
does it matter that these will return `messages.BlockProposal` instead of the actual type `*messages.BlockProposal`?

Reply:
No because this value is only used for logs